Signal/src/ViewControllers/ConversationView/MessagesViewController.m

```diff
@@ -485,8 +485,17 @@ - (void)configureForThread:(TSThread *)thread
     _callOnOpen = callOnViewAppearing;
 
     [self.uiDatabaseConnection beginLongLivedReadTransaction];
-    self.messageMappings =
-        [[YapDatabaseViewMappings alloc] initWithGroups:@[ thread.uniqueId ] view:TSMessageDatabaseViewExtensionName];
+
+    if (thread.uniqueId.length > 0) {
+        self.messageMappings = [[YapDatabaseViewMappings alloc] initWithGroups:@[ thread.uniqueId ]
+                                                                          view:TSMessageDatabaseViewExtensionName];
+    } else {
+        OWSFail(@"uniqueId unexpectedly empty for thread: %@", thread);
+        self.messageMappings =
+            [[YapDatabaseViewMappings alloc] initWithGroups:@[] view:TSMessageDatabaseViewExtensionName];
+        return;
```

Comment:
This is happening in production!?  That is, someone is somehow navigating into conversation view for an unsaved thread?

Reply:
Maybe a better fix is to save the thread if its unsaved (e.g. has not uniqueId)?

Reply:
I'm definitely leaning towards the "save the thread" approach - want to discuss? There's other places in this class that assume the thread is saved / has a uniqueId.

Reply:
> This is happening in production!?

Not often, but yes. 

> That is, someone is somehow navigating into conversation view for an unsaved thread?

No, I don't think that's the cause, since thread id's are assigned in init, not on save. Honestly not sure what the issue is, but hopefully the new logging will make it clearer if people find themselves in this state.

Reply:
I wonder if thread itself is nil?